ykrt/src/compile/jitc_yk/jit_ir.rs

```diff
@@ -317,10 +337,34 @@ impl Instruction {
         }
     }
 
-    /// Returns the size (in bytes) of the value that this instruction generates.
-    pub(crate) fn def_abi_size(&self) -> usize {
-        debug_assert!(self.is_def());
-        8 // FIXME
+    /// Returns the type of the local variable that the instruction defines (if any).
+    pub(crate) fn def_type<'a>(&self, m: &'a Module) -> Option<&'a Type> {
+        match self {
+            Self::Load(li) => Some(li.type_(m)),
+            Self::LoadGlobal(..) => todo!(),
+            Self::LoadArg(..) => {
+                // FIXME: This is nonsense, but we can't todo!() right now or we can't test
+                // anything.
+                Some(&Type::Ptr)
+            }
+            Self::Call(..) => todo!(),
+            Self::PtrAdd(..) => Some(&Type::Ptr),
+            Self::Store(..) => None,
+            Self::StoreGlobal(..) => None,
+        }
+    }
+
+    /// Returns the size of the local variable that this instruction defines (if any).
+    pub(crate) fn def_byte_size(&self, m: &Module) -> Option<usize> {
```

Comment:
Is returning `None` ever useful? Is this more a case of "panic if the user was stupid enough to call this on a non-definition"?

Reply:
We could just panic I suppose.

Reply:
Yes, I would `panic` myself.